bin/firefox-driver.js

```diff
@@ -1,38 +1,14 @@
 "use strict";
 
-const webdriver = require('selenium-webdriver'),
-    firefox = require('selenium-webdriver/firefox'),
-    By = webdriver.By,
-    until = webdriver.until,
-    Key = webdriver.Key,
-    child = require("child_process"),
-    express = require("express"),
-    bodyParser = require("body-parser");
+const webdriver = require('selenium-webdriver');
+const firefox = require('selenium-webdriver/firefox');
+const By = webdriver.By;
+const until = webdriver.until;
+const Key = webdriver.Key;
 
 function firefoxBinary() {
-  /**
-   * Binary.prototype.launch is being overriden to change
-   * child.spawn to child.exec.
-   *
-   * Unfortunately, it also removes linux support because that
-   * would have included additional dependencies.
-   *
-   * [BUG 2059](https://github.com/SeleniumHQ/selenium/issues/2059)
-   */
-  function launch(profile) {
-    var env = {};
-    Object.assign(env, this.env_, {XRE_PROFILE_PATH: profile});
-    var args = ['-foreground'].concat(this.args_);
-    return this.locate().then(function(firefox) {
-      return child.exec(firefox + " " + args.join(" "), {
-        env: env
-      });
-    });
-  }
-
-  var binary = new firefox.Binary('/Applications/Firefox.app/Contents/MacOS/firefox-bin');
-  binary.addArguments('--start-debugger-server 6080')
-  binary.launch = launch;
+  var binary = new firefox.Binary();
+  binary.addArguments('--start-debugger-server', '6080')
```

Comment:
I'm able to kill the launch logic because arguments are passed in key and value as opposed to just option.